src/Core/Coordinator.cpp

```diff
@@ -1,21 +1,41 @@
 #include "Coordinator.h"
 
-Coordinator::Coordinator()
-{
-};
+#include <iostream>
+#include <ostream>
+
+Coordinator::Coordinator(){};
 
 void Coordinator::init()
 {
     m_componentManager = std::make_unique<ComponentManager>();
     m_entityManager = std::make_unique<EntityManager>();
     m_systemManager = std::make_unique<SystemManager>();
+    m_serverEntityManager = std::make_unique<ServerEntityManager>();
```

Comment:
czemu powstał serverEntityManager, czym to się rózni od zwykłego entityManagera i czemu powstały dwa równoległe ECS?

Reply:
ServerEntityManger to klasa działająca dokładnie tak samo jak EntityManager, ale przeznaczona do mapowania entity z gry na id z serwera i vice versa. Miałem problem w jaki sposób zrobić wspólny id dla przeciwników u obu graczy. Uznałem, że to będzie najbardziej eleganckie rozwiązanie. Sam ServerEntityManager nie ma pełnić funkcji ECS, ma jedynie mapować obiekty, które przychodzą z serwera na obiekty w grze, żeby obiekt na serwerze był zawsze tym samym obiektem u każdego z graczy. Docelowo miały być tak mapowane wszystkie obiekty generowane przez serwer.